mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -0,0 +1,349 @@
+        # skip the test for non-supported layouts.
+        # MIOpenDriver currently only support NCHW.
+        nanoSeconds = 0
+    output(config.generateCSVContent(nanoSeconds), outputFile)
+
+def printPerformance(mlirFileName, miopenFileName):
+    with open(mlirFileName, 'r') as mlirOutput:
+        mlirResults = [line[:-1] for line in mlirOutput.readlines()]
+ 
+    with open(miopenFileName, 'r') as miopenOutput:
+        miopenFlops = [','+line.rstrip().split(',')[-1] for line in miopenOutput.readlines()]
+
+    results = [ i+j for i, j in zip(mlirResults, miopenFlops)]
+    header = ConvConfiguration.generateCSVHeader()+',MIOpenTFlops'
+    
+    table = PrettyTable()
+    table.field_names = header.split(',')
+    for res in results:
+        table.add_row(res.split(','))
+    print(table)
+
+    with open("MLIR_vs_MIOpen.html", 'w') as htmlOutput:
+        htmlOutput.write(table.get_html_string(format=True, 
+                                               attributes={'border': 1, 
+                                                           'style': 'border-width:1px; border-collapse: collapse; background-color: #dddddd;'}))
+ 
+def generatePerformanceResults(configs, xdlops):
+    mlirFileName = date.today().strftime("mlir.%m%d%y")
+    with open(mlirFileName, 'w') as mlirOutput:
+        for testVector in configs:
+            benchmarkMLIR(testVector.split(sep=' '), mlirOutput, xdlops)
+
+    miopenFileName = date.today().strftime("miopen.%m%d%y")
+    with open(miopenFileName, 'w') as miopenOutput:
+        for testVector in configs:
+            benchmarkMIOpen(testVector.split(sep=' '), miopenOutput, xdlops)
+
+    printPerformance(mlirFileName, miopenFileName)     
+
+# Main function.
+if __name__ == '__main__':
+    """ 
+usage examples: 
+  python3 MIOpenDriver.py
+  python3 MIOpenDriver.py -o mlir.perf -b
+  python3 MIOpenDriver.py -o miopen.perf -bmiopen
+  python3 MIOpenDriver.py -o mlir.perf conv -F 1 -f NCHW -I NCHW -O NCHW -n 256 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1
+  python3 MIOpenDriver.py -o miopen.perf -miopen conv -F 1 -f NCHW -I NCHW -O NCHW -n 256 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1
+    """    
+
+    xdlops = False
+    configs = getConfigurations(configurationFileName);
+    r = subprocess.run(f'/opt/rocm/bin/rocm_agent_enumerator -t GPU|grep gfx908', shell=True)
+    if r.returncode == 0:
+        xdlops = True
+
+    if len(sys.argv) == 1:
+        # batch benchmark with MLIR and MIOpen.
+        generatePerformanceResults(configs, xdlops)
+    else:
+        if sys.argv[1] == '-o':
+            fileName = sys.argv[2]
+            sys.argv.pop(1)
+            sys.argv.pop(1)
+        else:
+            fileName = date.today().strftime("perf.%m%d%y")
+
+        with open(fileName, 'w') as outputFile:
+            output(ConvConfiguration.generateCSVHeader(), outputFile)
+            if sys.argv[1] == '-b':
```

Comment:
To expand on the comment from the previous review, this options parsing code is very inflexible. It assumes, for example, that `-o` will be listed first (and it also uses that weird X11 convention of `-single-dash-long-options`). On the other hand, we're the only ones that'll be calling this. On the other, some of this could surprise people. Have you considered a `getopt` implementation instead - they'll gather up all the command line arguments that aren't recognized as arguments (such as the single config being tested, if we're doing a one-config test) but parse `argv` more flexibly.

Reply:
Cool, thanks. that makes sense for me.